paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java

```diff
@@ -45,46 +48,42 @@ public class CatalogController {
     @Autowired private CatalogService catalogService;
 
     /**
-     * Create a filesystem catalog.
+     * Create a catalog.
      *
-     * @param catalogInfo The catalogInfo for the filesystem catalog.
+     * @param catalogInfo The catalogInfo for the catalog.
      * @return The created catalog.
      */
-    @PostMapping("/createFilesystemCatalog")
-    public R<Void> createFilesystemCatalog(@RequestBody CatalogInfo catalogInfo) {
+    @PostMapping("/createCatalog")
+    public R<Void> createCatalog(@RequestBody CatalogInfo catalogInfo) {
         if (!catalogService.checkCatalogNameUnique(catalogInfo)) {
             return R.failed(Status.CATALOG_NAME_IS_EXIST, catalogInfo.getCatalogName());
         }
 
         try {
-            CatalogCreator.createFilesystemCatalog(catalogInfo.getWarehouse());
+            if (catalogInfo.getCatalogType().equalsIgnoreCase(CatalogMode.FILESYSTEM.getMode())) {
+                PaimonServiceFactory.createFileSystemCatalogService(
+                        catalogInfo.getCatalogName(), catalogInfo.getWarehouse());
+            } else if (catalogInfo.getCatalogType().equalsIgnoreCase(CatalogMode.HIVE.getMode())) {
+                if (StringUtils.isNotBlank(catalogInfo.getHiveConfDir())) {
+                    PaimonServiceFactory.createHiveCatalogService(
+                            catalogInfo.getCatalogName(),
+                            catalogInfo.getWarehouse(),
+                            catalogInfo.getHiveUri(),
+                            catalogInfo.getHiveConfDir());
+                } else {
+                    PaimonServiceFactory.createHiveCatalogService(
+                            catalogInfo.getCatalogName(),
+                            catalogInfo.getWarehouse(),
+                            catalogInfo.getHiveUri(),
+                            null);
+                }
+            } else {
+                PaimonServiceFactory.createFileSystemCatalogService(
```

Comment:
Could this branch merge with `if (catalogInfo.getCatalogType().equalsIgnoreCase(CatalogMode.FILESYSTEM.getMode()))`?